synapse/events/spamcheck.py

```diff
@@ -125,3 +125,25 @@ def user_may_publish_room(self, userid, room_id):
             return True
 
         return self.spam_checker.user_may_publish_room(userid, room_id)
+
+    def check_username_for_spam(self, userid: str, display_name: str) -> bool:
+        """Checks if a user ID or display name are considered "spammy" by this server.
+
+        If the server considers a username spammy, then it will not be included in
+        user directory results.
+
+        Args:
+            userid: The ID of the user to check
+            display_name: The display name of the user to check
```

Comment:
We should keep this even if Mjolnir won't be using it, but instead can we instead send over a profile object? Someone might want to check on the display name, avatar, and whatever else we add in the future.

Reply:
That makes sense, yes. It is a simple `dict`, not a class unfortunately.

Reply:
a `dict` would be fine, I think. Just make sure it's frozen so the spam checker doesn't modify profiles :p

Reply:
I made this change. There is no `frozendict` in Python, I made a (shallow) copy, which should be OK since both the keys and values are strings.